src/pages/workspaces/workspace/analysis/AzureComputeModal.js

```diff
@@ -0,0 +1,324 @@
+  //     diskSize: currentRuntimeDetails.diskConfig?.size || defaultAzureDiskSize,
+  //     region: currentRuntimeDetails.runtimeConfig?.region || defaultAzureRegion
+  //   } : {}
+  // }
+
+  const doesRuntimeExist = () => !!currentRuntimeDetails
+
+  const renderActionButton = () => {
+    const commonButtonProps = {
+      tooltipSide: 'left',
+      disabled: Utils.cond([viewMode === 'deleteEnvironment', () => getIsRuntimeBusy(currentRuntimeDetails)],
+        () => doesRuntimeExist()),
+      tooltip: Utils.cond([viewMode === 'deleteEnvironment', () => getIsRuntimeBusy(currentRuntimeDetails) ? 'Cannot delete a runtime while it is busy' : undefined],
+        [doesRuntimeExist(), () => 'Update not supported for azure runtimes'],
+        () => undefined)
+    }
+
+    return h(ButtonPrimary, {
+      ...commonButtonProps,
+      onClick: () => applyChanges()
+    }, [
+      Utils.cond(
+        [viewMode === 'deleteEnvironment', () => 'Delete'],
+        [doesRuntimeExist(), () => 'Update'],
+        () => 'Create'
+      )
+    ]
+    )
+  }
+
+  // Helper functions -- begin
+  const applyChanges = _.flow(
+    Utils.withBusyState(setLoading),
+    withErrorReportingInModal('Error modifying cloud environment', onDismiss)
+  )(async () => {
+    //TODO: metrics onclick
+    //sendCloudEnvironmentMetrics()
+
+    //each branch of the cond should return a promise
+    await Utils.cond([viewMode === 'deleteEnvironment', () => Ajax().Runtimes.runtimeV2(workspaceId, currentRuntimeDetails.runtimeName).delete()], //delete runtime
+      [Utils.DEFAULT, () => {
+        const disk = {
+          size: computeConfig.diskSize,
+          //We do not currently support re-attaching azure disks
+          name: Utils.generatePersistentDiskName(),
+          labels: { saturnWorkspaceNamespace: namespace, saturnWorkspaceName: workspaceName }
+        }
+
+        return Ajax().Runtimes.runtimeV2(workspaceId, Utils.generateRuntimeName()).create({
+          region: computeConfig.region,
+          machineSize: computeConfig.machineType,
+          labels: {
+            saturnWorkspaceNamespace: namespace,
+            saturnWorkspaceName: workspaceName
+          },
+          disk
+        })
+      }]
+    )
+
+    onSuccess()
+  })
+
+  const renderMainForm = () => {
+    return h(Fragment, [
+      div({ style: { padding: '1.5rem', borderBottom: `1px solid ${colors.dark(0.4)}` } }, [
+        renderTitleAndTagline(),
+        renderCostBreakdown()
+      ]),
+      div({ style: { padding: '1.5rem', overflowY: 'auto', flex: 'auto' } }, [
+        renderApplicationConfigurationSection(),
+        renderComputeProfileSection(),
+        renderBottomButtons()
+      ])
+    ])
+  }
+
+  //TODO this does not actually compute cost as is, see IA-3348
+  //It is possible that once we compute the cost, we would like to parameterize this and make it a shared function between the equivalent in ComputeModal
```

Comment:
Perhaps don't even show this section until you can display cost information?

Reply:
I like it there for style / spacing purposes personally